README.md

```diff
@@ -473,22 +485,193 @@ When the scanner successfully decodes a `TransactionRequest` it will call its de
+    
+**Usage**
+```kotlin
+socketClient.setChannelListener(object : SocketChannelCallback {
+    override fun onJoinedChannel(topic: SocketTopic) {
+        // Do something
+    }
+
+    override fun onLeftChannel(topic: SocketTopic) {
+        // Do something
+    }
+
+    override fun onError(apiError: APIError) {
+        // Handle an error
+    }
+})
+```
+
+## Listen for the custom event
+
+The custom event is the special event that currently limited to the `TransactionRequest` and the `TransactionConsumption` event.
+All custom events will be a sub-class of the `SocketCustomEventCallback`. `SocketCustomEventCallback` is the **required** generic callback that you will need to pass its sub-class when joining to the channel for listening for the event.
+All possible callbacks are the following:
+
+### TransactionRequest Event
+
+When creating a `TransactionRequest` that requires a confirmation it is possible to listen for all incoming using the `TransactionRequestCallback`.
+The possible events are: 
+* `onTransactionConsumptionRequest(TransactionConsumption)`: Invoked when a `TransactionConsumption` is trying to consume the `TransactionRequest`. 
+This allows the requester to [confirm](https://github.com/omisego/ios-sdk#confirm-a-transaction-consumption) or not the consumption if legitimate. 
+* `onTransactionConsumptionFinalizedSuccess(TransactionConsumption)`: Invoked if a `TransactionConsumption` has been finalized successfully, and the transfer was made between the 2 addresses.
+* `onTransactionConsumptionFinalizedFail(APIError)`: Invoked if a `TransactionConsumption` fails to consume the request.
+        
+### TransactionConsumption Event
+ 
+Similarly to the `TransactionRequestCallback`, a `TransactionConsumption` can be listened for incoming confirmations using the `TransactionConsumptionCallback`.
+The possible events are:
+* `onTransactionConsumptionFinalizedSuccess(TransactionConsumption)`: Invoked if a `TransactionConsumption` has been finalized successfully, and the transfer was made between the 2 addresses.
+* `onTransactionConsumptionFinalizedFail(APIError)`: Invoked if a `TransactionConsumption` fails to consume the request.
+
+**Usage**
+```kotlin
+// The transaction requestor listen for the event 
+transactionRequest.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionRequestCallback() {
+   override fun onTransactionConsumptionRequest(transactionConsumption: TransactionConsumption) {
+       // Do something
+   }
+
+   override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+       // Do something
+   }
+
+   override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+       // Do something
+   }
+})
+
+// The transaction consumer listen for the event
+transactionConsumption.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionConsumptionCallback() {
```

Comment:
If you have 2 transactions pending waiting for consumption, will you receive only the one for the transaction it's applied to?

Reply:
If you are the consumer and you `consume` 2 `TransactionRequest`, then you'll get 2 `TransactionConsumption`. If you want to receive confirm events for both transaction, you need to call it like this.

```kotlin
// This receives after you consume `transactionRequest1`
transactionConsumption1.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionConsumptionCallback())

// This receives after you consume `transactionRequest2`
transactionConsumption2.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionConsumptionCallback())
```